modules/svg/mapilio_images.js

```diff
@@ -254,31 +299,47 @@ export function svgMapilioImages(projection, context, dispatch) {
             .append('g')
             .attr('class', 'markers');
 
-        layer = layerEnter
-            .merge(layer);
+        layer = layerEnter.merge(layer);
 
         if (enabled) {
-            if (service && ~~context.map().zoom() >= minZoom) {
-                editOn();
-                update();
-                service.loadImages(projection);
-                service.loadLines(projection);
+            let zoom = ~~context.map().zoom();
+            if (service) {
+                if (zoom >= imageMinZoom) {
+                    editOn();
+                    update();
+                    service.loadImages(projection);
+                    service.loadLines(projection, zoom);
+                } else if (zoom >= lineMinZoom) {
+                    editOn();
+                    update();
+                    service.loadImages(projection);
+                    service.loadLines(projection, zoom);
+                    service.hideViewer(context);
+                } else {
+                    editOff();
+                    dispatch.call('photoDatesChanged', this, 'mapilio', []);
+                    service.selectImage(context, null);
+                    service.hideViewer(context);
+                }
             } else {
-                dispatch.call('photoDatesChanged', this, 'mapilio', []);
                 editOff();
             }
         } else {
             dispatch.call('photoDatesChanged', this, 'mapilio', []);
+            service.hideViewer(context);
```

Comment:
What's the intention here? I think the previous behavior was the intended one: keep the viewer open in case the user just wants to temporarily hide the sequences and photo markers.